linked_list/main.py

```diff
@@ -44,16 +45,19 @@ class Node(object):
     """
 
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem 
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        if self.elem == other.elem:
+            return True
+        return False
 
     def __repr__(self):
-        pass
+        return str(self)
```

Comment:
Usually __repr__ is indicated for a more "debug" message. Something like "<Node(3)>".